docs/en/advanced_guides/customize_runtime.md

```diff
@@ -1,17 +1,217 @@
+
+### Step 1: Create a new hook
+
+Depending on your intention of this hook, you need to implement corresponding functions according to the hook point of your expectation.
+
+For example, if you want to modify the value of a hyper-parameter according to the training iter and two other hyper-parameters after every train iter, you could implement a hook like:
+
+```python
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Optional, Sequence
+
+from mmengine.hooks import Hook
+from mmengine.model import is_model_wrapper
+
+from mmseg.registry import HOOKS
+from mmseg.utils import get_model
+
+
+@HOOKS.register_module()
+class NewHook(Hook):
+    """Docstring for NewHook.
+    """
+
+    def __init__(self, a: int, b: int) -> None:
+        self.a = a
+        self.b = b
+
+    def before_train_iter(self,
+                          runner,
+                          batch_idx: int,
+                          data_batch: Optional[Sequence[dict]] = None) -> None:
+        cur_iter = runner.iter
+        # Get model if the input model is a model wrapper
+        if is_model_wrapper(runner.model):
+          model = runner.model.module
+        model.hyper_parameter = self.a * cur_iter + self.b
+```
+
+### Step 2: Import the new hook
+
+Then we need to ensure `NewHook` imported. Assuming `NewHook` is in `mmseg/engine/hooks/new_hook.py`, modify `mmseg/engine/hooks/__init__.py` as below
+
+```python
+...
+from .new_hook import NewHook
+
+__all__ = [..., NewHook]
+```
+
+### Step 3: Modify the config
+
+```python
+custom_hooks = [
+    dict(type='NewHook', a=a_value, b=b_value)
+]
+```
+
+You can also set the priority of the hook as below:
+
+```python
+custom_hooks = [
+    dict(type='NewHook', a=a_value, b=b_value, priority='ABOVE_NORMAL')
+]
+```
+
+By default, the hook's priority is set as `NORMAL` during registration.
+
+### Modify default runtime hooks
+
+There are some common hooks that are not registered through `custom_hooks`, they are:
 
 ```python
-optimizer = dict(type='Adam', lr=0.0003, weight_decay=0.0001)
+default_hooks = dict(
+    timer=dict(type='IterTimerHook'),
+    logger=dict(type='LoggerHook', interval=50, log_metric_by_epoch=False),
+    param_scheduler=dict(type='ParamSchedulerHook'),
+    checkpoint=dict(type='CheckpointHook', by_epoch=False, interval=2000),
+    sampler_seed=dict(type='DistSamplerSeedHook'),
+    visualization=dict(type='SegVisualizationHook'))
 ```
 
-To modify the learning rate of the model, the users only need to modify the `lr` in the config of optimizer. The users can directly set arguments following the [API doc](https://pytorch.org/docs/stable/optim.html?highlight=optim#module-torch.optim) of PyTorch.
+Here we reveals how what we can do with `logger` and `checkpoint`.
+
+#### Checkpoint config
+
+The MMCV runner will use `checkpoint_config` to initialize [`CheckpointHook`](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/checkpoint_hook.py#L19).
```

Comment:
why `checkpoint=dict(type='CheckpointHook', by_epoch=False, interval=2000)` in default hooks

but `checkpoint_config` is used to initialize `CheckpointHook`